### DISCUSSION TITLE
InlineQueryResultPhoto: Pass image bytes instead of URL

### DISCUSSION THREAD

**Discussion by xk4rim** [eternnoir/pyTelegramBotAPI#1557] [Q&A] [answered]
Hello,

I'd like to pass an image object instead of a URL to [InlineQueryResultPhoto](https://github.com/eternnoir/pyTelegramBotAPI/blob/756775027698bf975cab387746f810b199d17e79/telebot/types.py#L1801).
As I saw in the docs, [InlineQueryResultPhoto ](https://github.com/eternnoir/pyTelegramBotAPI/blob/756775027698bf975cab387746f810b199d17e79/telebot/types.py#L1801)takes photo_url and thumb_url. I'd like to know if there's a way I can pass bytes instead of URLs.

Thank you.

**Comment by xk4rim** [comment]
I found a solution for photo_url  by passing an [InputMediaPhoto](https://github.com/eternnoir/pyTelegramBotAPI/blob/5d74e18d1a14d378dc2b47c9fea5ce0936219f99/telebot/types.py#L2654) to input_message_content parameter.
However, I couldn't find anything for thumb_url.

**Comment by Badiboy** [comment]
It's the limitation of the Telegram.
https://core.telegram.org/bots/api#inlinequeryresultphoto
Thumb can be passed only by URL.

PS: I also do not thing _input_message_content_ will help you to send Photo...
https://core.telegram.org/bots/api#inputmessagecontent

**Comment by xk4rim** [comment, marked as answer]
Thank you, @Badiboy. I discovered that _input_message_content_ requires a URL yesterday and forgot to update this question.
After hours of digging, I finally found a workaround:

1. Send the photo to a private channel made solely for caching.
2. Retrieve the photo to answer the inline query.
3. Delete the photo in the caching channel.

```python
msg = bot.send_photo(chat_id=CACHE_CHANNEL_ID, photo=image)
photo_file_id = msg.photo[0].file_id

option = InlineQueryResultCachedPhoto(
                    id='0',
                    title=title,
                    description=description,
                    photo_file_id=photo_file_id,
                    caption=caption,
                    parse_mode='MARKDOWN'
       )

bot.answer_inline_query(
            query.id,
            [option],
            cache_time=60*15
)

bot.delete_message(
            chat_id=CACHE_CHANNEL_ID,
            message_id=msg.message_id,
)

```

[Reference](https://stackoverflow.com/a/61399534)
